node/src/test_utils/recorder_counter_msgs.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) 2025, MASQ (https://masq.ai) and/or its affiliates. All rights reserved.
+
+#![cfg(test)]
+
+use crate::test_utils::recorder_stop_conditions::{ForcedMatchable, MsgIdentification};
+use actix::{Message, Recipient};
+use std::any::TypeId;
+use std::cell::RefCell;
+use std::collections::hash_map::Entry;
+use std::collections::HashMap;
+
+pub trait CounterMsgGear: Send {
+    fn try_send(&self);
+}
+
+pub struct SendableCounterMsgWithRecipient<Msg>
+where
+    Msg: Message + Send,
+    Msg::Result: Send,
+{
+    msg_opt: RefCell<Option<Msg>>,
+    recipient: Recipient<Msg>,
+}
+
+impl<Msg> CounterMsgGear for SendableCounterMsgWithRecipient<Msg>
+where
+    Msg: Message + Send,
+    Msg::Result: Send,
+{
+    fn try_send(&self) {
+        let msg = self.msg_opt.take().unwrap();
+        self.recipient.try_send(msg).unwrap()
+    }
+}
+
+impl<Msg> SendableCounterMsgWithRecipient<Msg>
+where
+    Msg: Message + Send + 'static,
+    Msg::Result: Send,
+{
+    pub fn new(msg: Msg, recipient: Recipient<Msg>) -> SendableCounterMsgWithRecipient<Msg> {
+        Self {
+            msg_opt: RefCell::new(Some(msg)),
+            recipient,
+        }
+    }
+}
+
+pub struct SingleCounterMsgSetup {
+    // Leave them private
+    trigger_msg_type_id: TypeId,
+    condition: MsgIdentification,
+    // Multiple messages sent off on reaction are allowed
+    // (Imagine a message handler whose execution goes about more than just one msg dispatch)
+    msg_gears: Vec<Box<dyn CounterMsgGear>>,
+}
+
+impl SingleCounterMsgSetup {
+    pub fn new(
+        trigger_msg_type_id: TypeId,
+        trigger_msg_id_method: MsgIdentification,
+        counter_messages: Vec<Box<dyn CounterMsgGear>>,
+    ) -> Self {
+        Self {
+            trigger_msg_type_id,
+            condition: trigger_msg_id_method,
+            msg_gears: counter_messages,
+        }
+    }
+}
+
+#[derive(Default)]
+pub struct CounterMessages {
+    msgs: HashMap<TypeId, Vec<SingleCounterMsgSetup>>,
+}
+
```

Comment:
You'd like a type alias to represent `TypeId` in the `HashMap`. You suggested to use: `TriggerMsgTypeId`.